apps/src/applab/firebaseCounters.js

```diff
@@ -110,13 +113,48 @@ export function incrementRateLimitCounters() {
 }
 
 /**
+ * Returns a promise which resolves once Firebase constants have been fetched from
+ * the server and cached locally. Also starts listening for future changes to constants.
+ * @returns {Promise<>}
+ */
+function loadConstants() {
+  if (tableRowCountLimit && rateLimitMap) {
+    // The firebase constants have already been loaded.
+    return Promise.resolve();
+  }
+
+  const constantsRef = getConstants();
+  return constantsRef.once('value', snapshot => {
+    handleLoadConstants(snapshot.val());
+
+    // Make sure we don't listen multiple times.
+    constantsRef.off();
+
+    // Update globals to reflect firebase constants any time they change in the future.
+    constantsRef.on('value', snapshot => handleLoadConstants(snapshot.val()));
+  }).catch((error) => {
+    // Failed to load constants. Clear the promise so they can be fetched again.
+    return Promise.reject(error);
+  });
```

Comment:
good catch Brent (bad `catch` Dave). This was left over from a more complicated thing that I had tried and abandoned.